src/request/order.ts

```diff
@@ -0,0 +1,40 @@
+import isURL = require('validator/lib/isURL');
+import { ArgumentError } from '../errors';
+
+interface OrderProps {
+  amount?: number;
```

Comment:
I'm wondering if `number` is a sufficient type. The Java lib has this property as a `BigDecimal`. I wonder if we can find a type that is equivalent, as it seems to be a bit more restrictive.